magefile.go

```diff
@@ -8,42 +8,197 @@ package main
 
 import (
 	"fmt"
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
 	"os"
 	"path/filepath"
 
 	"github.com/magefile/mage/sh"
+
+	"github.com/elastic/package-registry/util"
 )
 
 var (
-	buildDir     = "./build"
-	publicDir    = filepath.Join(buildDir, "public")
 	packagePaths = []string{"./packages"}
-	tarGz        = true
 )
 
-func Build() error {
-	err := os.RemoveAll(publicDir)
+type fieldEntry struct {
+	name  string
+	aType string
+}
+
+func Validate() error {
```

Comment:
Why not use the dry-run in the registry for this?

Reply:
Does `dry-run` verify required fields?

Reply:
I would need to check. But if not, we should add the check?

Reply:
I think we reuse the `dry-run` in all projects. It will become single source of knowledge/validation.

Reply:
++

Reply:
https://github.com/elastic/package-registry/pull/507